margin_max: 5
margin_min: 0.5
precise_alignment_height: 2
cruise_height: 8